appengine/flexible/Pubsub/Pubsub.Sample/Program.cs

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+using Google.Cloud.PubSub.V1;
+using Microsoft.Extensions.Options;
+using System.Net;
+
+namespace Pubsub;
+
+public class Program
+{
+    private WebApplication App {get; set;}
+
+    private Program(string[] args)
+    {
+        var builder = WebApplication.CreateBuilder(args);
+        var options = new PubsubOptions()
+        {
+            ProjectId = builder.Configuration["TEST_PROJECT_ID"],
+            VerificationToken = builder.Configuration["TEST_VERIFICATION_TOKEN"],
+            TopicId = builder.Configuration["TEST_TOPIC_ID"],
+            SubscriptionId = builder.Configuration["TEST_SUBSCRIPTION_ID"],
+            AuthSubscriptionId = builder.Configuration["TEST_AUTH_SUBSCRIPTION_ID"],
+            ServiceAccountEmail = builder.Configuration["TEST_SERVICE_ACCOUNT_EMAIL"]
+        };
+        var topic = CreateTopicAndSubscription(options);
+        builder.Services.AddSingleton(Options.Create(options));
+        builder.Services.AddSingleton(PublisherClient.Create(topic));
+        builder.Services.AddMvc();
+        App = builder.Build();
+        App.UseStaticFiles();
+        App.MapControllerRoute(
+            name: "default",
+            pattern: "{controller=Home}/{action=Index}/{id?}");
+    }
+
+    static TopicName CreateTopicAndSubscription(PubsubOptions options)

```

Comment:
Note, this and the next method are both lifted from Startup.cs. I could refactor if it makes sense.

Reply:
The error handling in particular seems a little off. You know better what the goal of this sample is, but I guess these methods can be made to be cleaner (and clearer).

Reply:
I ended up actually *deleting* these, because the public docs have you create these resources anyway (so no need to do them in code too, I feel like): https://cloud.google.com/appengine/docs/flexible/writing-and-responding-to-pub-sub-messages?tab=.net